### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by GitItOmar**

### CODE REVIEW

**Comment by GitItOmar** [comment]
Please remove unused functions and files from the frontend and web folder.

File: web/index.js [hunk 1]
```diff
@@ -85,6 +64,7 @@ cron.schedule("0 0 * * *", async () => {
   console.log("Running cron job to fetch, store, and update product prices");
   try {
     const session = await getSession();
+    // strange API
     await fetchStoreAndUpdatePrices(
       { locals: { shopify: { session } } },
       {
```
Comment (GitItOmar) [inline web/index.js:70]: Improve strange API

File: web/controllers/priceController.js [hunk 2]
```diff
@@ -138,6 +139,7 @@ export const fetchStoreAndUpdatePrices = async (req, res) => {
 
       const existingMetafield = metafieldResponse.data.product.metafield;
 
+      // no conditional needed. You can create and update with the same operation
       // Step 2: Update or create the metafield
       if (existingMetafield) {
         // Update existing metafield
```
Comment (GitItOmar) [inline web/controllers/priceController.js:145]: No conditional needed here. You can update and create with the same operation.

File: web/controllers/priceController.js [hunk 3]
```diff
@@ -107,6 +107,7 @@ export const fetchStoreAndUpdatePrices = async (req, res) => {
     // Extract prices from products and prepare data for insertion
     const prices = products.map((product) => ({
       productId: product.id,
+      // this is not the lowest price
       price: parseFloat(product.variants.edges[0].node.price),
```
Comment (GitItOmar) [inline web/controllers/priceController.js:111]: This is not the lowest variant price but the first variant price.

**Comment by GitItOmar** [comment]
Regarding the frontend folder, we will procide you with an interface, you can implement to make manual actions

**Comment by GitItOmar** [comment]
Please use eslint with airbnb as style guide. Please note that eslint just released v9. v9 is not compatible with the Airbnb style guide. Please use v8.

**Comment by GitItOmar** [comment]
The retry mechanisms and logging still has to be implemented. You can use winston and log to a file until we find a fitting solution for production.

File: web/controllers/priceController.js (hunk 3 above)
Comment (yousefomar724) [reply web/controllers/priceController.js:18]: Yea, because I sorted the prices ascending, the first variant price is the lowest

**Pull request merged**